app/src/main/java/com/keylesspalace/tusky/components/timeline/viewmodel/CachedTimelineRemoteMediator.kt

```diff
@@ -53,11 +53,19 @@ class CachedTimelineRemoteMediator(
 
         try {
             var dbEmpty = false
+
+            val topPlaceholderId = if (loadType == LoadType.REFRESH) {
+                timelineDao.getTopPlaceholderId(activeAccount.id)
+            } else {
+                null // don't execute the query if it is not needed
+            }
+
             if (!initialRefresh && loadType == LoadType.REFRESH) {
                 val topId = timelineDao.getTopId(activeAccount.id)
                 topId?.let { cachedTopId ->
                     val statusResponse = api.homeTimeline(
                         maxId = cachedTopId,
+                        sinceId = topPlaceholderId, // so already existing placeholders don't get accidentally overwritten
```

Comment:
So it will load up from which point? Sorry I'm a bit lost. As far as I understand since topmost placeholder. It might be confusing if it's near the top I guess because we might refresh only the small part but at your discretion.

Reply:
Yes, it will only refresh above the topmost placeholder. This will almost always be a full page, because for less statuses to be loaded one must have been deleted. Yes it may be weird in case a lot of statuses have been deleted but I think that is such a rare case that it is acceptable.

Reply:
You are right, this should be very rare.